sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -205,3 +205,19 @@ case class UnresolvedExtractValue(child: Expression, extraction: Expression)
 
   override def toString: String = s"$child[$extraction]"
 }
+
+case class UnresolvedAlias(child: Expression) extends NamedExpression
```

Comment:
Can we reuse the `Alias`? And it will be unresolved if name is null or empty.

Just a suggestion, probably @marmbrus can make the final decision.